soc/silabs/silabs_s2/Kconfig

```diff
@@ -10,4 +10,7 @@ config SOC_FAMILY_SILABS_S2
 
 rsource "*/Kconfig"
 
+config HW_HAS_SILABS_DCDC
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_SILABS_SERIES2_DCDC))
```

Comment:
please avoid these redundant Kconfig options and just use DT natively in C code

Reply:
I had been pointed to https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/common/Kconfig.peripherals in the past, but if this is a pattern that we don't want to propagate, and just use DT directly instead, that is completely fine by me :)